.github/workflows/build.image.yml

```
@@ -0,0 +1,56 @@
+name: Build image
+
+on:
+    push:
+        branches:
+            - main
+        paths:
+            - 'requirements/*'
+            - 'docker/dockerfile'
+    pull_request:
+        branches:
+            - main
+        paths:
+            - 'requirements/*'
+            - 'docker/dockerfile'
+
+jobs:
+
+    set-vars:
+        uses: ./.github/workflows/set-vars.yml
+
+    build-image:
+        needs: [set-vars]
+        runs-on: [ubuntu-latest]
+        container:
+            image: docker:latest
+        permissions:
+            packages: write
+        steps:
+            - name: Checkout code.
+              uses: actions/checkout@v4
+
+            - name: Set up QEMU.
+              uses: docker/setup-qemu-action@v3
+
+            - name: Set up Docker Buildx.
+              uses: docker/setup-buildx-action@v3
+
+            - name: Login to GitHub Container Registry.
+              uses: docker/login-action@v3
+              with:
+                  registry: ${{ needs.set-vars.outputs.CI_REGISTRY }}
+                  username: ${{ github.actor }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Build and push.
+              uses: docker/build-push-action@v5
+              with:
+                  context: .
+                  platforms:  linux/amd64,linux/arm64
+                  push:       ${{ github.ref == 'refs/heads/main' }}
+                  file:       docker/dockerfile
+                  tags:       ${{ needs.set-vars.outputs.CI_IMAGE }}
+                  cache-from: type=registry,ref=${{ needs.set-vars.outputs.CI_IMAGE }}
+                  cache-to:   type=inline
+                  labels:     "org.opencontainers.image.source=${{ github.repositoryUrl }}"
```

I think we usually indent like this
```suggestion
                  platforms : linux/amd64,linux/arm64
                  push      : ${{ github.ref == 'refs/heads/main' }}
                  file      : docker/dockerfile
                  tags      : ${{ needs.set-vars.outputs.CI_IMAGE }}
                  cache-from: type=registry,ref=${{ needs.set-vars.outputs.CI_IMAGE }}
                  cache-to  : type=inline
                  labels    : "org.opencontainers.image.source=${{ github.repositoryUrl }}"
```

---

OK. Aligned it this way :paw_prints: